lib/widgets/menu/menu_custom_navbar.dart

```diff
@@ -0,0 +1,114 @@
+  static const int savePage = 2;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: EdgeInsets.fromLTRB(
+        MediaQuery.of(context).size.width * paddingWidth,
+        MediaQuery.of(context).size.width * paddingHeight,
+        MediaQuery.of(context).size.width * paddingWidth,
+        MediaQuery.of(context).size.width * paddingHeight,
+      ),
+      height: MediaQuery.of(context).size.width * height,
+      decoration: BoxDecoration(
+        color: Theme.of(context).colorScheme.primary,
+        boxShadow: [
+          BoxShadow(
+            color: Colors.black.withOpacity(boxShadowOpacity),
+            blurRadius: boxBlurRadius,
+            offset: const Offset(
+              boxOffsetX,
+              boxOffsetY,
+            ),
+          ),
+        ],
+        borderRadius: BorderRadius.circular(navbarRadius),
+      ),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+        children: [
+          IconButton(
+            onPressed: () {
+              setState(
+                () {
+                  widget.currentIndex.setIndex(homePage);
+                  widget.pageController.animateToPage(
+                    homePage,
+                    duration: const Duration(milliseconds: 500),
+                    curve: Curves.fastLinearToSlowEaseIn,
+                  );
+                },
+              );
+            },
+            icon: Icon(
+              Icons.home_filled,
+              color: colors.background.withOpacity(
+                  widget.currentIndex.getIndex == homePage ? 1 : opacity),
+            ),
+          ),
+          IconButton(
+            onPressed: () {
+              setState(
+                () {
+                  widget.currentIndex.setIndex(searchPage);
+                  widget.pageController.animateToPage(
+                    searchPage,
+                    duration: const Duration(milliseconds: 500),
+                    curve: Curves.fastLinearToSlowEaseIn,
+                  );
+                },
+              );
+            },
+            icon: Icon(
+              Icons.search_rounded,
+              color: colors.background.withOpacity(
+                  widget.currentIndex.getIndex == searchPage ? 1 : opacity),
+            ),
+          ),
+          IconButton(
+            onPressed: () {
+              setState(
+                () {
+                  widget.currentIndex.setIndex(savePage);
+                  widget.pageController.animateToPage(
+                    savePage,
+                    duration: const Duration(milliseconds: 500),
+                    curve: Curves.fastLinearToSlowEaseIn,
+                  );
+                },
+              );
+            },
+            icon: Icon(
+              Icons.bookmark_added,
+              color: colors.background.withOpacity(
+                  widget.currentIndex.getIndex == savePage ? 1 : opacity),
+            ),
+          ),
```

Comment:
All this widgets have almost the same functionality, you can create a new method that return an IconButton

Reply:
You're right, done it